src/sync/remote_mongo_client.hpp

```diff
@@ -0,0 +1,53 @@
+////////////////////////////////////////////////////////////////////////////
+//
+// Copyright 2020 Realm Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or utilied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+////////////////////////////////////////////////////////////////////////////
+
+#ifndef CORE_REMOTE_MONGO_CLIENT_HPP
+#define CORE_REMOTE_MONGO_CLIENT_HPP
+
+#include <string>
+#include <map>
+#include <realm/util/optional.hpp>
+#include "stitch_service_client.hpp"
+
+namespace realm {
+namespace app {
+
+class RemoteMongoDatabase;
+
+/// A client responsible for communication with the Stitch API
+class RemoteMongoClient {
+public:
+
+    RemoteMongoClient(StitchServiceClient service) : m_service(service) { }
```

Comment:
[minor] Remove all references to `Stitch`. Should be `RealmServiceClient`.

Reply:
Done